src/service.ts

```diff
@@ -18,28 +18,52 @@
  * @module commonGrpc/service
  */
 
-'use strict';
-
+import {DecorateRequestOptions, Service, ServiceConfig, ServiceOptions, util} from '@google-cloud/common';
+import {replaceProjectIdToken} from '@google-cloud/projectify';
+import {loadSync, PackageDefinition, ServiceDefinition} from '@grpc/proto-loader';
 import * as duplexify from 'duplexify';
 import * as extend from 'extend';
+import * as grpc from 'grpc';
 import * as is from 'is';
+import {Response} from 'request';
 import * as retryRequest from 'retry-request';
-import {Service, util, ServiceConfig} from '@google-cloud/common';
-import {replaceProjectIdToken} from '@google-cloud/projectify';
+import {Duplex} from 'stream';
 import * as through from 'through2';
-import * as grpc from 'grpc';
-import {loadSync, PackageDefinition, ServiceDefinition} from '@grpc/proto-loader';
+
+// tslint:disable-next-line no-any
+const g = global as any;
+
+export interface GrpcResponse {
+  code: number;
+  message: string;
+}
+
+export interface GrpcServiceOptions extends ServiceOptions {
+  maxRetries?: number;
+}
 
 interface GrpcOptions {
   deadline?: Date;
 }
 
+export interface GrpcStatus {
+  code?: number;
+}
+
+export interface ProtoOptions {
+  service?: string;
+  method?: string;
+  timeout?: number;
+  retryOpts?: retryRequest.Options;
+  stream?: Duplex;
+}
+
 /**
  * Configuration object for GrpcService.
  */
 export interface GrpcServiceConfig extends ServiceConfig {
   /** Metadata to send with every request. */
-  grpcMetadata: grpc.Metadata;
+  grpcMetadata: grpc.Metadata&{[index: string]: Function};
```

Comment:
This pattern is repeated lots. Declare the following type at the top, and use it here.

```ts
interface Bag<T> {
  [key: string]: T;
}
```

Reply:
Good call!